kakao/KakaoLoginIOS/KakaoLoginIOS/MainViewController.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  MainViewController.swift
+//  KakaoLoginIOS
+//
+//  Created by Kevin Kang on 2017. 9. 18..
+//  Copyright © 2017년 Hara Kang. All rights reserved.
+//
+import FirebaseAuth
+
+class MainViewController: UIViewController {
+    @IBOutlet weak var profileImageView: UIImageView!
+    @IBOutlet weak var emailTextView: UITextView!
+    @IBOutlet weak var displayNameTextView: UITextView!
+    @IBOutlet weak var navigationBar: UINavigationBar!
+    @IBOutlet weak var navigationTitleItem: UINavigationItem!
+    
+    override func viewDidLoad() {
+        let user = Auth.auth().currentUser
+        if user == nil {
+            self.dismiss(animated: true, completion: {
+            })
+        } else {
+            updateUserProfileUI(user: user!)
+        }
```

Comment:
This should be moved out of `viewDidLoad` and into `viewDidAppear`.  Otherwise, the view controller may attempt to dismiss itself when it's not presented or the controller may be presented, populate with the user's info, be dismissed, then be presented again later, but not refresh the user info.

The conditional itself should use Swift `if let` optional binding to avoid the force-unwrap later.

```swift
if let user = Auth.auth().currentUser {
  updateUserProfile(user: user)
} else {
  self.dismiss(animated: true, completion: {})
}
```

This method is also missing a call to super.